lib/internal/modules/cjs/loader.js

```diff
@@ -879,6 +879,46 @@ function createRequire(filename) {
 
 Module.createRequire = createRequire;
 
+Module.containsModuleSyntax = (source) => {
+  // Detect whether input source code contains at least one `import` or `export`
+  // statement. This can be used by dependent utilities as a way of detecting ES
+  // module source code from Script/CommonJS source code. Since our detection is
+  // so simple, we can avoid needing to use Acorn for a full parse; we can
+  // detect import or export statements just from the tokens. Also as of this
+  // writing, Acorn doesn't support import() expressions as they are only Stage
+  // 3; yet Node already supports them.
+  const acorn = require('internal/deps/acorn/acorn/dist/acorn');
+  source = stripShebang(source);
+  source = stripBOM(source);
```

Comment:
These should likely be switched around.

Reply:
I’m happy to switch them; I was following the style in `check_syntax.js`:

https://github.com/nodejs/node/blob/7c816b7588f0e2910bcef5db48f71296cd3b67c2/lib/internal/main/check_syntax.js#L53-L74

So if it’s supposed to be `stripBOM` first, they should probably be switched there too.

Reply:
@GeoffreyBooth probably yes - a BOM comes before shebang syntax when parsing text. Please switch them around.

Reply:
note that a shebang after a bom is not actually a shebang. you should use the `stripShebangOrBOM` function which correctly handles that

Reply:
if the source is a module we shouldn't be stripping the bom, as we don't strip BOM for es module text anywhere else.